### ISSUE TITLE
Patch-based inpainting doesn't work with 32-bit images

### ISSUE THREAD

**Issue by Beep6581** [dtschump/gmic#36]
Hello

Using GIMP 2.9.8 (commit 18794a6ba2) and G'MIC-Qt 2.1.5, when I open a 32-bit image (doesn't matter whether fp or int), fill a rectangle with #ff0000 and attempt to use G'MIC's patch-based inpainting to fix the area, nothing changes.

Changing the image to 8-bit precision in GIMP makes G'MIC detect the mask correctly.

Test file:
https://filebin.net/u5iiznksd49dvpht/gmic_inpaint_patch-based.xcf

**Comment by dtschump** [comment]
Hello. Thanks for reporting.
This patch https://github.com/dtschump/gmic/commit/e2b47190182dbc5091d04bb0c2b2755cc0d44179 should fix this issue, when applying the filter.
Seems like it does not work for the preview though, this requires some further investigation.

**Comment by Beep6581** [comment]
Will this fix be part of version 2.1.9?

**Comment by dtschump** [comment]
Yes, exactly. Should be actually working with latest builds 2.1.9_pre available on the G'MIC website, and after pressing the 'filter update' button.

**Comment by Beep6581** [comment]
Thank you.
